### PR TITLE
Delay setting converter in RepositoryRestConfig

### PR DESCRIPTION
**Pull request by gregturn**

### CODE REVIEW

File: src/main/java/mihalcin/customconverter/RepositoryRestConfig.java [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 
-import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.core.convert.support.ConfigurableConversionService;
 import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurerAdapter;
+import org.springframework.stereotype.Component;
 
-//@Configuration
-@RequiredArgsConstructor
+@Component
```
Comment (pmihalcin) [inline src/main/java/mihalcin/customconverter/RepositoryRestConfig.java:8]: Should it be component or configuration?

**Review by pmihalcin — APPROVED** [review]

**Comment by gregturn** [comment]
To answer your question, `@Configuration` classes typically carry `@Bean` definitions. This class is TURNED INTO a bean, so `@Component` is the proper annotation.

**Comment by pmihalcin** [comment]
Thanks for clarification..

When I try to move `TranslationStringToSpecificationStatusEnumConverter` to subpackage e.g. `convert`, it fails on the same failure as before.
Why is it necessary to keep both `TranslationStringToSpecificationStatusEnumConverter` and `RepositoryRestConfig` under same package?

**Comment by gregturn** [comment]
Spring Boot will the package and all sub packages where the launching app is. If you moved it outside that hierarchy then it won’t be found.

**Comment by pmihalcin** [comment]
I moved it to sub-package. That's why I'm surprised. I pushed one commit here: https://github.com/pmihalcin/custom-converter-in-spring-data-rest/commit/127a4983d92c7cc5013748aa828c27fcae6370b2
It's strange that it behaves that way.

**Pull request merged**